File: app/models/match.rb
class Match < ActiveRecord::Base
  has_and_belongs_to_many :teams
  
  def winner
    self.teams.find { |t| t.id == self.winner_id }
  end
  
  def winner=(team)
    self.winner_id = team.id
    self.save
  end
  
  def as_json options
    {
      id: self.id,
      teams: self.teams,
      winner: self.winner_id
    }
  end
  
end

File: app/assets/javascripts/brakkit/models/match.js

//= require ./record
//= require ./team

Brakkit.Match = Brakkit.Record.extend({
  teams : null,
  winner : null,
  round_id : null,
  urlRoot : '/matches',
  attributes : function(){
    var self = this,
        team_ids = self.get('teams').getEach('id');
    var _attributes = {
      match : { winner_id : self.winner },
      round_id : self.get('round_id'),
      teams : team_ids
    }
    return _attributes;
  }.property('winner','teams')
});